turbot/resource_turbot_resource.go

```diff
@@ -203,36 +271,35 @@ func resourceTurbotResourceImport(d *schema.ResourceData, meta interface{}) ([]*
 	return []*schema.ResourceData{d}, nil
 }
 
-func buildDataUpdateProperties(d *schema.ResourceData, properties []interface{}) (map[string]interface{}, error) {
-	var err error
-	// convert data from json string to map
-	var dataMap map[string]interface{}
-	dataString := d.Get("data").(string)
-	if dataMap, err = helpers.JsonStringToMap(dataString); err != nil {
-		return nil, fmt.Errorf("error build resource mutation input, failed to unmarshal data: \n%s\nerror: %s", dataString, err.Error())
-	}
-	for _, element := range properties {
-		if _, ok := dataMap[element.(string)]; ok {
-			delete(dataMap, element.(string))
+func buildDataUpdateProperties(data map[string]interface{}, forbiddenProperties []interface{}) map[string]interface{} {
+	for _, element := range forbiddenProperties {
+		if _, ok := data[element.(string)]; ok {
+			delete(data, element.(string))
 		}
 	}
-
-	return dataMap, nil
+	return data
 }
 
 func buildResourceInput(d *schema.ResourceData, properties []interface{}) (map[string]interface{}, error) {
 	var err error
+	var ok bool
+	var dataString, metaDataString interface{}
 	input := mapFromResourceData(d, properties)
-	// convert data from json string to map
-	dataString := d.Get("data").(string)
-	if input["data"], err = helpers.JsonStringToMap(dataString); err != nil {
-		return nil, fmt.Errorf("error build resource mutation input, failed to unmarshal data: \n%s\nerror: %s", dataString, err.Error())
+	if dataString, ok = d.GetOk("data"); !ok {
```

Comment:
// check whether either. "data" or "full_data" properties are set and retrieve value